scripts/fast-reboot

```diff
@@ -70,19 +88,59 @@ docker exec -i bgp killall -9 bgpd
 docker kill lldp > /dev/null
 
 # Kill teamd, otherwise it gets down all LAGs
+# Note: teamd must be killed before syncd, because it will send the last packet through CPU port
+# TODO: stop teamd gracefully to allow teamd to send last valid update to be sure we'll have 90 seconds reboot time
 docker kill teamd > /dev/null
 
-# syncd graceful stop is supported only for Broadcom platforms only for now
-if [[ "$sonic_asic_type" = 'broadcom' ]];
-then
-    # Gracefully stop syncd
-    docker exec -i syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
-
-    # Check that syncd was stopped
-    while docker top syncd | grep -q /usr/bin/syncd
-    do
-          sleep 0.1
-    done
+if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
+    # Freeze orchagent for warm restart
+    # Note: assume that 5 seconds is enough for orchagent to process the request and freeze, otherwise we treat the system as non-warm-reboot-able
+    docker exec -i swss /usr/bin/orchagent_restart_check -w 5000 || {
+        echo "RESTARTCHECK failed\n" >&2
+        exit 10
+    }
+    # Set whole system warm reboot flag
+    config warm_restart enable system
+fi
+# Kill swss dockers
+docker kill swss
+
+# Warm reboot: dump state to host disk
+# Note: even if syncd changed ASIC_DB before killed, we don't care
+if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
+    # Dump redis content to directory
+    # Note: don't use pretty mode redis-dump (1.1) since there is a known bug with key pattern
+    DUMP_CMD="redis-dump -s /var/run/redis/redis.sock"
+    WARM_DIR=/host/warmboot
+    mkdir -p $WARM_DIR
+    # Note: requiring redis-dump-load
+    # Save applDB in /host/warm-reboot/appl_db.json
+    $DUMP_CMD -d 0 -o $WARM_DIR/appl_db.json
+    # Save configDB in /host/warm-reboot/config_db.json
+    $DUMP_CMD -d 4 -o $WARM_DIR/config_db.json
+    # Save stateDB (only FDB_TABLE and WARM_RESTART_TABLE) in /host/warm-reboot/state_db.json
+    # WARNING WARNING WARNING: a trick to dump both FDB_TABLE|* and WARM_RESTA*
+    # TODO: replace it with readable mechanism to dump multiple key patterns into one single json file
+    $DUMP_CMD -d 6 -k "[FW][DA][BR][_M][T_][AR][BE][LS][ET][|A]*" -o $WARM_DIR/state_db.json
+    # Save asicDB in /host/warm-reboot/asic_db.json
+    $DUMP_CMD -d 1 -o $WARM_DIR/asic_db.json
+fi
+
+if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
+    # Gracefully stop syncd for warm-reboot
+    systemctrl stop syncd
```

Comment:
Is this a typo 'systemctrl' -> 'systemctl'?
```suggestion
    systemctl stop syncd
```

Reply:
Yes. Thanks!

Reply:
Yes. Thanks!

---
In reply to: [226010381](https://github.com/Azure/sonic-utilities/pull/338#discussion_r226010381) [](ancestors = 226010381)